source/IrcDotNet/IrcClient.cs

```diff
@@ -1458,20 +1468,36 @@ public virtual void Disconnect()
 
         protected void ParseMessage(string line)
         {
+            IDictionary<string, string> tags = null;
+            string lineAfterTags = null;
             string prefix = null;
             string lineAfterPrefix = null;
 
+            // Extract tags from message, if it contains any
+            if (line[0] == '@')
+            {
+                // tags are separated to the rest of the message with a space
+                var endTagsIndex = line.IndexOf(' ');
+                Debug.Assert(endTagsIndex != -1);
```

Comment:
I'm not familiar with the entire codebase here, So maybe this is a known choice - But we should consider not having asserts in production code. If we need to handle specific use cases here, Let's let the code handle it rather than dying with an assert.

Reply:
Agree on asserts.

Reply:
to answer this thread, should I try to change all asserts over to some exception, or just logging and drop the message? Because there are two preexisting asserts in this method of the same character: failing for corrupted or malicious irc lines.

EDIT: Oh, my previous comment was pending for over 2 years

Reply:
I merged it but this still possibly worth fixing.